sbom/examples/container_image/ubi9-micro-container-9.4-6.1716471860.spdx.json

```diff
@@ -0,0 +1,2251 @@
+  "packages": [
+    {
+      "SPDXID": "SPDXRef-image-index",
+      "name": "ubi9-micro-container",
+      "versionInfo": "9.4-6.1716471860",
+      "supplier": "Organization: Red Hat",
+      "downloadLocation": "NOASSERTION",
+      "licenseConcluded": "NOASSERTION",
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referenceType": "purl",
+          "referenceLocator": "pkg:oci/ubi-micro@sha256%3A1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d?repository_url=registry.access.redhat.com/ubi9&tag=9.4-6.1716471860"
+        },
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referenceType": "purl",
+          "referenceLocator": "pkg:oci/ubi9-micro@sha256%3A1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d?repository_url=registry.access.redhat.com&tag=9.4-6.1716471860"
+        }
+      ],
+      "checksums": [
+        {
+          "algorithm": "SHA256",
+          "checksumValue": "1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d"
+        }
+      ]
+    },
+    {
+      "SPDXID": "SPDXRef-ubi9-micro-container-s390x",
+      "name": "ubi9-micro-container_s390x",
+      "versionInfo": "9.4-6.1716471860",
+      "supplier": "Organization: Red Hat",
+      "downloadLocation": "NOASSERTION",
+      "licenseConcluded": "NOASSERTION",
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referenceType": "purl",
+          "referenceLocator": "pkg:oci/ubi-micro@sha256%3A1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d?arch=s390x&repository_url=registry.access.redhat.com/ubi9&tag=9.4-6.1716471860"
+        },
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referenceType": "purl",
+          "referenceLocator": "pkg:oci/ubi9-micro@sha256%3A1c8483e0fda0e990175eb9855a5f15e0910d2038dd397d9e2b357630f0321e6d?arch=s390x&repository_url=registry.access.redhat.com&tag=9.4-6.1716471860"
+        }
+      ],
+      "checksums": [
+        {
+          "algorithm": "SHA256",
+          "checksumValue": "sha256:2c9e70f4174747c6b53d253e879177c52731cc4bdc5fe9c6a2555412d849a952"
+        }
+      ]
+    },
+    {
+      "SPDXID": "SPDXRef-s390x-libgcc",
+      "name": "libgcc",
+      "versionInfo": "11.4.1",
+      "supplier": "Organization: Red Hat",
+      "downloadLocation": "NOASSERTION",
+      "packageFileName": "libgcc-11.4.1-3.el9.s390x.rpm",
+      "licenseConcluded": "NOASSERTION",
+      "externalRefs": [
+        {
+          "referenceCategory": "PACKAGE-MANAGER",
+          "referenceType": "purl",
+          "referenceLocator": "pkg:rpm/redhat/libgcc@11.4.1?arch=s390x&repository_url=rhel-9-for-s390x-baseos-rpms"
+        },
+        {
+          "referenceCategory": "OTHER",
+          "referenceType": "sbom_ref",
+          "referenceLocator": "https://access.redhat.com/security/data/sbom/v1/rpm/gcc-0:11.4.1-3.el9.spdx.json.bz2"
+        }
```

Comment:
I suppose this `sbom_ref` is a non-normative way to say "you can find the SBOM for that over here"?

The full external document linking style would be this: https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#6.6
where we'd have things like `"relatedSpdxElement": "DocumentRef-rpm-gcc:SPDXRef-s390x-libgcc"` in `"relationships"` to describe relationships to packages not listed in this document.

Reply:
And, to be clear, I think we don't necessarily want to use that full external document linking style, if purl matching is sufficient and provides more consumer convenience.

Reply:
> I suppose this `sbom_ref` is a non-normative way to say "you can find the SBOM for that over here"?
> 

Yep!

> The full external document linking style would be this: https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#6.6 where we'd have things like `"relatedSpdxElement": "DocumentRef-rpm-gcc:SPDXRef-s390x-libgcc"` in `"relationships"` to describe relationships to packages not listed in this document.

Oh neat, I didn't realize this existed, but it does create a lot more objects in the SBOM since you need a separate `DocumentRef` and relationship object for each RPM.

The CycloneDX approach also seems to be similar to the one I used here where there is a reference type called `bom` to point to other SBOMs: https://cyclonedx.org/use-cases/#external-references

Do you have a preference for one or the other?

> And, to be clear, I think we don't necessarily want to use that full external document linking style, if purl matching is sufficient and provides more consumer convenience.

purl matching works when you import it into something like Guac/Trustification, but if we are to also provide these documents as is for anyone to download and parse on their own, we'd be asking them to also re-implement the matching of purls themselves to discover the links between different artifacts. Providing the reference(s) in the document would make that a lot easier, I'd think.

Ideally, our VEX and advisory files would also link / refer to the SBOMs so that we could potentially reduce their size to just be one or more references to actual SBOM files instead of replicating the information between the two.

Reply:
> Providing the reference(s) in the document would make that a lot easier, I'd think.

+1

Reply:
I removed the `sbom_ref` reference since we'll be relying on purl matching only.